# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/l10n/tests/mochitest/dom_localization/test_l10n_mutations.html
# Commit: 595529cd906f
# Full Hash: 595529cd906f94cdb2bc12a7a40072db0c07198e
# Author: Zibi Braniecki <zbraniecki@mozilla.com>
# Date: 2021-08-03 22:01:19
# Regressor Bug: 1613705
# File Overlap Count: 1
# Description:
#   Bug 1613705 - [localization] part9: Update DOMLocalization to match Localization constructor. r=nika,emilio
#   
#   Depends on D113691
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D113692
# ==============================================================================

diff -r 9f3e2963d925 -r 595529cd906f dom/l10n/tests/mochitest/dom_localization/test_l10n_mutations.html
--- a/dom/l10n/tests/mochitest/dom_localization/test_l10n_mutations.html	Tue Aug 03 05:52:03 2021 +0000
+++ b/dom/l10n/tests/mochitest/dom_localization/test_l10n_mutations.html	Tue Aug 03 05:52:03 2021 +0000
@@ -7,12 +7,15 @@
   <link rel="stylesheet" type="text/css" href="chrome://mochikit/content/tests/SimpleTest/test.css">
   <script type="application/javascript">
   "use strict";
-  async function* generateBundles(resourceIds) {
-    const bundle = new FluentBundle("en-US");
-    bundle.addResource(new FluentResource("title = Hello World"));
-    bundle.addResource(new FluentResource("title2 = Hello Another World"));
-    yield bundle;
-  }
+  const l10nReg = new L10nRegistry();
+  const fs = [
+    { path: "/localization/en-US/mock.ftl", source: `
+title = Hello World
+title2 = Hello Another World
+` },
+  ];
+  const source = L10nFileSource.createMock("test", ["en-US"], "/localization/{locale}", fs);
+  l10nReg.registerSources([source]);
 
   window.onload = async function() {
     SimpleTest.waitForExplicitFinish();
@@ -20,12 +23,13 @@
     const domLoc = new DOMLocalization(
       [],
       false,
-      { generateBundles },
+      l10nReg,
+      ["en-US"],
     );
 
     const h1 = document.querySelectorAll("h1")[0];
 
-    domLoc.addResourceIds(["dummy.ftl"]);
+    domLoc.addResourceIds(["/mock.ftl"]);
     domLoc.connectRoot(document.body);
 
     await domLoc.translateRoots();